src/components/CheckEditor/CheckEditor.tsx

```diff
@@ -43,30 +44,41 @@ const getStyles = (theme: GrafanaTheme) => ({
   `,
 });
 
-export const CheckEditor: FC<Props> = ({ check, instance, onReturn }) => {
+export const CheckEditor: FC<Props> = ({ check, onReturn }) => {
+  const {
+    instance: { api },
+  } = useContext(InstanceContext);
   const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const { alertRules, setRulesForCheck, deleteRulesForCheck } = useAlerts(check?.id);
   const styles = useStyles(getStyles);
-
   const defaultValues = useMemo(() => getDefaultValuesFromCheck(check), [check]);
 
   const formMethods = useForm<CheckFormValues>({ defaultValues, mode: 'onChange' });
   const selectedCheckType = formMethods.watch('checkType').value as CheckType;
 
   const isEditor = hasRole(OrgRole.EDITOR);
 
-  const { execute: onSubmit, error, loading: submitting } = useAsyncCallback(async (values: CheckFormValues) => {
-    const updatedCheck = getCheckFromFormValues(values, defaultValues);
-    if (check?.id) {
-      await instance.updateCheck({
-        id: check.id,
-        tenantId: check.tenantId,
-        ...updatedCheck,
-      });
-    } else {
-      await instance.addCheck(updatedCheck);
+  const { execute: onSubmit, error, loading: submitting } = useAsyncCallback(
+    async ({ alert, ...checkValues }: CheckFormValues) => {
+      const updatedCheck = getCheckFromFormValues(checkValues, defaultValues);
+      if (check?.id) {
+        await api?.updateCheck({
+          id: check.id,
+          tenantId: check.tenantId,
+          ...updatedCheck,
+        });
+        if (alert) {
+          await setRulesForCheck(check.id, alert, checkValues.job, checkValues.target);
+        }
+      } else {
+        const { id } = await api?.addCheck(updatedCheck);
+        if (alert) {
+          await setRulesForCheck(id, alert, checkValues.job, checkValues.target);
+        }
```

Comment:
You could remove a bit of redundancy here by having `let id` before the `if`, so that you can have `await setRulesForCheck(id...)` just once after the `if`.